receiver/kafkareceiver/kafka_receiver.go

```diff
@@ -579,10 +593,29 @@ func (c *tracesConsumerGroupHandler) ConsumeClaim(session sarama.ConsumerGroupSe
 			err = c.nextConsumer.ConsumeTraces(session.Context(), traces)
 			c.obsrecv.EndTracesOp(ctx, c.unmarshaler.Encoding(), spanCount, err)
 			if err != nil {
-				if c.messageMarking.After && c.messageMarking.OnError {
-					session.MarkMessage(message, "")
+				if errorRequiresBackoff(err) && c.backOff != nil {
+					backOffDelay := c.backOff.NextBackOff()
+					if backOffDelay == backoff.Stop {
+						if c.messageMarking.After && c.messageMarking.OnError {
+							session.MarkMessage(message, "")
+						}
+						return err
+					}
+					select {
+					case <-session.Context().Done():
+						return nil
+					case <-time.After(backOffDelay):
+						if !c.messageMarking.After {
+							// Unmark the message so it can be retried
+							session.ResetOffset(claim.Topic(), claim.Partition(), message.Offset, "")
+						}
```

Comment:
If `c.messageMarking.After == false`, message is already marked before the pipeline execution https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/37009/files#diff-8fa2a92bedeb4eca30c174611f6204e98b4d80570d7b0c37a3c8b2fdf4e75ef7L554-L556, reset the offset to the message offset so that it can be consumed again in the next for loop. Do not return the error which would interrupt the loop.

If `c.messageMarking.After == true` then do nothing